// unitags/dist/src/getUniswapv3LPTags.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
// Use dynamic import for node-fetch
import fetch from "node-fetch";
const GET_POOLS_QUERY = `
  query GetPools($lastTimestamp: Int) {
    pools(
      first: 1000,
      orderBy: createdAtTimestamp,
      orderDirection: asc,
      where: { createdAtTimestamp_gt: $lastTimestamp }
    ) {
      id
      createdAtTimestamp
      token0 {
        id
        name
        symbol
      }
      token1 {
        id
        name
        symbol
      }
    }
  }
`;
function returnTags() {
    return __awaiter(this, void 0, void 0, function* () {
        let lastTimestamp = 0;
        let allTags = [];
        let isMore = true;
        while (isMore) {
            console.log(typeof lastTimestamp);
            const response = yield fetch("https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3", {
                method: "POST",
                headers: {
                    "Content-Type": "application/json",
                    Accept: "application/json",
                },
                body: JSON.stringify({
                    query: GET_POOLS_QUERY,
                    variables: { lastTimestamp: lastTimestamp },
                }),
            });
            if (!response.ok) {
                throw new Error(`Network response was not ok: ${response.statusText}`);
            }
            const result = (yield response.json());
            const pools = result.data.pools;
            allTags.push(...transformPoolsToTags(pools));
            if (pools.length < 1000) {
                isMore = false;
            }
            else {
                lastTimestamp = parseInt(pools[pools.length - 1].createdAtTimestamp.toString());
            }
        }
        return allTags;
    });
}
function transformPoolsToTags(pools) {
    return pools.map((pool) => ({
        "Contract Address": pool.id,
        "Public Name Tag": `${pool.token0.symbol}/${pool.token1.symbol} pool`,
        "Project Name": "Uniswap v3",
        "UI/Website Link": "https://uniswap.org",
        "Public Note": `The liquidity pool contract on Uniswap v3 for the ${pool.token0.name}(${pool.token0.symbol}) / ${pool.token1.name}(${pool.token1.symbol}) pair.`,
    }));
}
export { returnTags };
